### PR TITLE
Return transformed coordinates when dragging in progress

### PR DESCRIPTION
**Pull request by kiRach**
Hi @w8r,

Thank you for your work and for your plugin.

This small PR contains little refactoring to extract method to allow transformation of single points according to the provided transformation matrix. I need this change to implement PR in your another plugin: https://github.com/w8r/Leaflet.draw.drag

UPDATE: I realized, that all my changes should be part of this plugin, not Leaflet.draw.drag. So I moved changes from https://github.com/w8r/Leaflet.draw.drag/pull/9 here.

Description from https://github.com/w8r/Leaflet.draw.drag/pull/9 still applies here. In short: now, when dragging in progress, when we call `getLatLng` method of `L.Circle` or `getLatLngs` of L.Polyline - we will get transformed points (according to transformation matrix). It allows us to get actual positions when dragging in progress.

Thanks,
Kirill

### CODE REVIEW

**Comment by kiRach** [comment]
Hi @w8r,

Did you have a chance to look through my changes?

File: src/Path.Drag.js [hunk 1]
```diff
@@ -171,18 +183,31 @@ L.Handler.PathDrag = L.Handler.extend( /** @lends  L.Path.Drag.prototype */ {
...
-      path._latlng = projection.unproject(
-        projection.project(path._latlng)._add(diff));
+      path._latlng = this._transformPoint(matrix, path._latlng);
       path._point._add(px);
     } else if (path._originalPoints) { // everything else
       for (i = 0, len = path._originalPoints.length; i < len; i++) {
-        latlng = path._latlngs[i];
-        path._latlngs[i] = projection
-          .unproject(projection.project(latlng)._add(diff));
+        path._latlngs[i] = this._transformPoint(matrix, path._latlngs[i]);
```
Comment (w8r) [inline src/Path.Drag.js]: see, I don't really like that, you see how many unnecessary transforms it introduces? maybe we can shift the diff to the arguments?

File: src/Path.Drag.js [hunk 2]
```diff
@@ -221,3 +244,33 @@ L.Path.prototype._initEvents = function() {
...
+  } else {
+    return this._getLatLng();
+  }
+};
+
+L.Polyline.prototype._getLatLngs = L.Polyline.prototype.getLatLngs;
+L.Polyline.prototype.getLatLngs = function() {
+  if (this.dragging && this.dragging.inProgress()) {
+    var points = this._getLatLngs();
+    points.forEach(function (point, i) {
```
Comment (w8r) [inline src/Path.Drag.js]: also benchmarks show that `forEach` is slower than the basic `for` loop

File: src/Path.Drag.js (hunk 1 above)
Comment (w8r) [inline src/Path.Drag.js]: 49ms vs 22 on my machine. Small numbers but you can see the impact. that might be the method call as well

Comment (kiRach) [inline src/Path.Drag.js]: Could you please provide me, how did you measure this?

I've just made a small benchmark and didn't notice difference. Actually, your original version is 0.005 longer in average in my tests ;)

######  my PR version

time |
------------
0.095ms |
0.098ms |
0.066ms |
0.054ms |
0.046ms |
0.055ms |
0.046ms |
0.055ms |
0.062ms |
0.055ms |
0.053ms |
0.055ms |
0.054ms |
0.053ms |
0.081ms |
0.051ms |
0.057ms |

`0.06094117647ms average`

######  original version

time |
--------
0.103ms |
0.084ms |
0.068ms |
0.057ms |
0.058ms |
0.056ms |
0.058ms |
0.049ms |
0.058ms |
0.059ms |
0.065ms |
0.063ms |
0.073ms |
0.057ms |
0.048ms |
0.058ms |
0.059ms |
0.088ms |

`0.0645ms average`

**Comment by w8r** [comment]
@kiRach is it per operation? Cause it hardly makes sense to me, there's an additional transform and function call per point, they may be very lightweight, but altogether they add up - that's what I've been checking. I just enabled timers around transformPoints function(to measure the whole operation) and dragged the heaviest polygon there is in the demo - that's how I got my numbers.

**Comment by w8r** [comment]
@kiRach, don't get me wrong, I'm not being an asshole here, just trying to make sure it's performant on large quantities of points. I appreciate your work, let's try and have a compromise here: if I understood correctly, the thing you need is to be able to get correctly transformed points __while__ dragging, right? Maybe let's not touch the `_transformPoints` method for now, and just adopt all the rest of your changes?

**Comment by w8r** [comment]
@kiRach, thanks!

**Comment by kiRach** [comment]
Hi @w8r, sorry for long reply - have been working on high priority hot fixes. I understand you concern and also want to find the solution what we both will be good with. I see that you've already merged this PR. Do you want me to open another one with fixes according your comments?

**Comment by w8r** [comment]
@kiRach, I edited your PR according to them already. I introduced some code repetition, which bothers me much less than the performance. I also just found an urgent need to use that functionality for one of my projects)))

**Pull request merged**